client/state/nps-survey/actions.js

```diff
@@ -32,10 +32,25 @@ export function forceNpsSurveyEligibility( isEligible ) {
 }
 
 export function setupNpsSurveyEligibility() {
-	// TODO: add calls to API once eligibility endpoint is in
-	return {
-		type: NPS_SURVEY_SETUP_ELIGIBILITY_REQUESTING,
-		isSessionPicked: 1 === random( 1, NPS_SURVEY_RAND_MAX ),
+	return ( dispatch ) => {
+		debug( 'Checking NPS eligibility...' );
+
+		if ( 1 === random( 1, NPS_SURVEY_RAND_MAX ) ) {
+			return wpcom
+				.undocumented()
+				.checkNPSSurveyEligibility()
+				.then( ( data ) => {
+					debug( '...Eligibility returned from endpoint.', data );
+					dispatch( forceNpsSurveyEligibility( data.display_survey ) );
```

Comment:
Interesting use of `forceNpsSurveyEligibility`... I hadn't really anticipated it would be used this way. Makes sense though. I would change two things though in the existing code, to make things more clear:

- Rename action `forceNpsSurveyEligibilty` to `setNpsSurveyEligibility`
- Rename action type `NPS_SURVEY_SETUP_ELIGIBILITY_REQUESTING` to `NPS_SURVEY_SET_ELIGIBILITY`